packages/mc-dev-authentication/middlewares/create-mc-dev-authentication-middleware.js

```diff
@@ -0,0 +1,76 @@
+const path = require('path');
+const pug = require('pug');
+const { logout } = require('../routes');
+
+const compileLoginView = pug.compileFile(
+  path.join(__dirname, '../views/login.pug')
+);
+const compileLogoutView = pug.compileFile(
+  path.join(__dirname, '../views/logout.pug')
+);
+
+function createMcDevAuthenticationMiddleware(applicationConfig) {
+  const htmlLogin = compileLoginView({ env: applicationConfig.env });
+  const htmlLogout = compileLogoutView({ env: applicationConfig.env });
+
+  const isDevAuthenticationMiddlewareDisabled =
+    String(applicationConfig.env.disableAuthRoutesOfDevServer) === 'true' ||
+    applicationConfig.env.servedByProxy;
+
+  /**
+   * @type {import('express').RequestHandler}
+   */
+  return (request, response, next) => {
+    if (request.originalUrl === '/api/graphql') {
+      response.statusCode = 404;
+      response.setHeader('Content-Type', 'application/json');
+      response.end(
+        JSON.stringify({
+          message: `This GraphQL endpoint is only available in production in the [Merchant Center Proxy Router](https://docs.commercetools.com/custom-applications/concepts/merchant-center-proxy-router). Please check that you are not calling this endpoint in development mode.`,
+        })
+      );
+      return;
+    }
+
+    if (applicationConfig.env.__DEVELOPMENT__?.oidc?.authorizeUrl) {
+      // Handle login page for OIDC workflow when developing against a local MC API.
+      if (
+        applicationConfig.env.__DEVELOPMENT__?.oidc?.authorizeUrl.startsWith(
+          'http://localhost'
+        )
+      ) {
+        if (request.originalUrl.startsWith('/login/authorize')) {
+          if (isDevAuthenticationMiddlewareDisabled) {
+            next();
+          } else {
+            response.end(htmlLogin);
+          }
+          return;
+        }
+      }
```

Comment:
Maybe I'm missing something but it seems here if we do have an `authorizeUrl` in the application configuration but it does not start with `http://localhost` or the `originalUrl` does not start with `/login/authorize` we will not call `next()` nor finish the `response` so we could end up with a hanging request.

Perhaps we should just let the request continue in those cases?
```suggestion
      if (
        applicationConfig.env.__DEVELOPMENT__?.oidc?.authorizeUrl.startsWith(
          'http://localhost'
        )
      ) {
        if (request.originalUrl.startsWith('/login/authorize')) {
          if (isDevAuthenticationMiddlewareDisabled) {
            next();
          } else {
            response.end(htmlLogin);
          }
          return;
        } else {
          next();
        }
      } else  {
        next();
      }
```

Reply:
In line L72 we do call `next()` as a fallback. Isn't that enough? 🤔

Reply:
Yes, it is.

Sorry.

Reply:
All good, thanks for double checking 🤗